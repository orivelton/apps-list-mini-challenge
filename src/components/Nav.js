import React, { useState, useEffect } from 'react';

const Nav = ({initialState, handleCategories, categoriesActive}) => {
  const [categories, setCategories] = useState([]);
  const [active, setActive] = useState(categoriesActive);

  useEffect(() => {
    setCategories([...new Set(initialState.map(({categories}) => categories).flat())]);
  }, [initialState]);

  useEffect(() => {
    setActive(categoriesActive)
  },[categoriesActive]);


  const handleClick = (e, item) => {
    setActive(item);
    e.preventDefault();
    handleCategories(item);
  }

  return (
    <nav className="nav-categories">
      <h2>Categories</h2>
      <ul className="nav-menu">
        {
          categories.length && categories.map(item => (
            <li key={item} >
              <button 
                href="/"
                className={active === item ? 'active' : ''}
                onClick={e => handleClick(e, item)}>
                {item}
              </button>
            </li>
          ))
        }
        {/* <li className="active"><a href="/">Channels</a></li> */}
      </ul>
    </nav>
  );
};

export default Nav;
