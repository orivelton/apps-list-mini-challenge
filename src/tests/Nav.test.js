import React from 'react';
import renderer from 'react-test-renderer';
import Nav from '../components/Nav';

test('renders correctly', () => {
  const tree = renderer.create(<Nav />).toJSON();
  expect(tree).toMatchSnapshot();
});
