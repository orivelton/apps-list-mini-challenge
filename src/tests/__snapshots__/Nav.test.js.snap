// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="nav-categories"
>
  <h2>
    Categories
  </h2>
  <ul
    className="nav-menu"
  />
</nav>
`;
